xds/src/main/java/io/grpc/xds/RbacFilter.java

```diff
@@ -0,0 +1,322 @@
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.protobuf.Any;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.Message;
+import io.envoyproxy.envoy.config.core.v3.CidrRange;
+import io.envoyproxy.envoy.config.rbac.v3.Permission;
+import io.envoyproxy.envoy.config.rbac.v3.Policy;
+import io.envoyproxy.envoy.config.rbac.v3.Principal;
+import io.envoyproxy.envoy.extensions.filters.http.rbac.v3.RBAC;
+import io.envoyproxy.envoy.extensions.filters.http.rbac.v3.RBACPerRoute;
+import io.grpc.Metadata;
+import io.grpc.ServerCall;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import io.grpc.Status;
+import io.grpc.xds.Filter.ServerInterceptorBuilder;
+import io.grpc.xds.internal.MatcherParser;
+import io.grpc.xds.internal.Matchers;
+import io.grpc.xds.internal.rbac.engine.GrpcAuthorizationEngine;
+import io.grpc.xds.internal.rbac.engine.GrpcAuthorizationEngine.AlwaysTrueMatcher;
+import io.grpc.xds.internal.rbac.engine.GrpcAuthorizationEngine.AndMatcher;
+import io.grpc.xds.internal.rbac.engine.GrpcAuthorizationEngine.AuthConfig;
+import io.grpc.xds.internal.rbac.engine.GrpcAuthorizationEngine.AuthDecision;
+import io.grpc.xds.internal.rbac.engine.GrpcAuthorizationEngine.AuthHeaderMatcher;
+import io.grpc.xds.internal.rbac.engine.GrpcAuthorizationEngine.AuthenticatedMatcher;
+import io.grpc.xds.internal.rbac.engine.GrpcAuthorizationEngine.DestinationIpMatcher;
+import io.grpc.xds.internal.rbac.engine.GrpcAuthorizationEngine.DestinationPortMatcher;
+import io.grpc.xds.internal.rbac.engine.GrpcAuthorizationEngine.InvertMatcher;
+import io.grpc.xds.internal.rbac.engine.GrpcAuthorizationEngine.Matcher;
+import io.grpc.xds.internal.rbac.engine.GrpcAuthorizationEngine.OrMatcher;
+import io.grpc.xds.internal.rbac.engine.GrpcAuthorizationEngine.PathMatcher;
+import io.grpc.xds.internal.rbac.engine.GrpcAuthorizationEngine.PolicyMatcher;
+import io.grpc.xds.internal.rbac.engine.GrpcAuthorizationEngine.RequestedServerNameMatcher;
+import io.grpc.xds.internal.rbac.engine.GrpcAuthorizationEngine.SourceIpMatcher;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+
+/** RBAC Http filter implementation. */
+final class RbacFilter implements Filter, ServerInterceptorBuilder {
+  private static final Logger logger = Logger.getLogger(RbacFilter.class.getName());
+
+  static final RbacFilter INSTANCE = new RbacFilter();
+
+  static final String TYPE_URL =
+          "type.googleapis.com/envoy.extensions.filters.http.rbac.v3.RBAC";
+
+  RbacFilter() {}
+
+  @Override
+  public String[] typeUrls() {
+    return new String[] { TYPE_URL };
```

Comment:
The RBAC filter should be registered for both "type.googleapis.com/envoy.extensions.filters.http.rbac.v3.RBAC" and "type.googleapis.com/envoy.extensions.filters.http.rbac.v3.RBACPerRoute". Otherwise, the XdsClient would fail to parse override RBAC filter configs. I will add a test case in #8228.